app/vmagent/remotewrite/client.go

```diff
@@ -481,3 +480,46 @@ again:
 }
 
 var remoteWriteRejectedLogger = logger.WithThrottler("remoteWriteRejected", 5*time.Second)
+
+// calculateRetryAfterDuration calculate the retry duration.
+// 1. Calculate next retry duration by backoff policy (x2) and max retry duration limit.
+// 2. use max(Retry-After duration, next retry duration).
+//
+// Also see: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6097
+func calculateRetryDuration(retryAfterDuration, retryDuration, maxRetryDuration time.Duration) time.Duration {
```

Comment:
Hey @jiekun @zekker6 

If server responds with exact Retry-After value, shouldn't we respect it as is? I mean, shouldn't we completely ignore default backoff policy? It seems like what users would expect to happen if they gone to extent of sending the specific Retry-After value

Reply:
Updated as mentioned above. I'd like to discuss: 
- Should we also respect the limit of `--remoteWrite.retryMinInterval` and `--remoteWrite.retryMaxTime`? So:
    1. If `RA` header exist, retry == `max(min(RA, retryMaxTime), retryMinInterval)`;
    2. If `RA` header not exist, use default back-off.

Reply:
I think if RA exists, client should respect it.
If not, fall back to the default behavior.